source/postprocess/velocity_boundary_statistics.cc

```diff
@@ -76,48 +80,78 @@ namespace aspect
                 // extract velocities
                 fe_face_values[this->introspection().extractors.velocities].get_function_values (this->get_solution(),
                     velocities);
-                // determine the max and min velocity on the face
+                // determine the max, min, and squared velocity on the face
+                // also determine the face area
                 double local_max = -std::numeric_limits<double>::max();
                 double local_min = std::numeric_limits<double>::max();
+                double local_sqvel = 0.0;
+                double local_fe_face_area= 0.0;
                 for (unsigned int q=0; q<fe_face_values.n_quadrature_points; ++q)
                   {
                     const double vel_mag = velocities[q].norm();
                     local_max = std::max(vel_mag,
                                          local_max);
                     local_min = std::min(vel_mag,
                                          local_min);
+                    local_sqvel += ((vel_mag * vel_mag) * fe_face_values.JxW(q));
+                    local_fe_face_area += fe_face_values.JxW(q);
                   }
-                // then merge them with the min/max velocities we found for other faces with the same boundary indicator
+
+                // then merge them with the min/max/squared velocities
+                // and face areas we found for other faces with the same boundary indicator
                 const types::boundary_id boundary_indicator
                   = cell->face(f)->boundary_id();
 
                 local_max_vel[boundary_indicator] = std::max(local_max,
                                                              local_max_vel[boundary_indicator]);
                 local_min_vel[boundary_indicator] = std::min(local_min,
                                                              local_min_vel[boundary_indicator]);
+                local_boundary_sqvel[boundary_indicator] += local_sqvel;
+                local_boundary_area[boundary_indicator] += local_fe_face_area;
               }
 
       // now communicate to get the global values
       std::map<types::boundary_id, double> global_max_vel;
       std::map<types::boundary_id, double> global_min_vel;
+      std::map<types::boundary_id, double> boundary_vel_rms;
       {
         // first collect local values in the same order in which they are listed
         // in the set of boundary indicators
         std::vector<double> local_max_values;
         std::vector<double> local_min_values;
+        std::vector<double> boundary_sqvel_values;
+        std::vector<double> boundary_area_values;
```

Comment:
I would call these `local_squared_velocity_values` and `local_boundary_area_values`

Reply:
Changed boundary_sqvel_values to local_velocity_square_integral_values to be more consistent with the map variable name.